pyccel/codegen/printing/ccode.py

```diff
@@ -2580,6 +2580,13 @@ def _print_ClassDef(self, expr):
         return methods + interfaces
 
     #================== List methods ==================
+    def _print_ListAppend(self, expr):
+        target = expr.list_obj
+        class_type = target.class_type
+        c_type = self.get_c_type(class_type)
+        arg = self._print(expr.args[0])
+        return f'{c_type}_push(&{target.name}, {arg});\n'
```

Comment:
You shouldn't use `&{target.name}`. What happens in any of the following situations:
1. `target` is an optional argument (so stored in a pointer already)
2. `target` is a class variable
3. `target` is an element of a list
4. etc

Reply:
```suggestion
        target = expr.list_obj
        class_type = target.class_type
        c_type = self.get_c_type(class_type)
        target_code = self._print(ObjectAddress(target))
        arg = self._print(expr.args[0])
        return f'{c_type}_push({target_code}, {arg});\n'
```

Reply:
Yes, you are right!
I should use this instead:
```python
    def _print_ListAppend(self, expr):
        target = expr.list_obj
        class_type = target.class_type
        c_type = self.get_c_type(class_type)
        arg = self._print(expr.args[0])
        list_obj = self._print(ObjectAddress(expr.list_obj))
        return f'{c_type}_push({list_obj}, {arg});\n'
```
right?

Reply:
Yes exactly

Reply:
oops i didn't see your other comment. Shall I go with yours or [this](https://github.com/pyccel/pyccel/pull/1982/commits/9331b105298b1d69a8273dfd246a6db1828f25e0)

Reply:
As you prefer. Yours is already committed so let's go with that

Reply:
I only gave a suggestion to be able to request changes ^^